### DISCUSSION TITLE
Do you miss additional Wallbox data?

### DISCUSSION THREAD

**Discussion by marq24** [marq24/ha-senec-v3#26] [General]
In the response of the SENEC device there are plenty of additional fields concerning the wallbox - since I do not own one (yet) I can't judge, if any of these additional fields might generated attentional value in HA... 

So is there anything "essential" missing? @toastbroat13 ?!

Here is the list of fields that would be available:
- ADDITIONAL_ERROR
- ALLOW_INTERCHARGE
- APPARENT_CHARGING_POWER
- BUS_ADR
- CS_ENABLED
- DETECTION_MODE
- ERROR_DETAILS
- **EV_CONNECTED**
- HW_TYPE
- **L1_CHARGING_CURRENT**
- **L1_USED**
- **L2_CHARGING_CURRENT**
- **L2_USED**
- **L3_CHARGING_CURRENT**
- **L3_USED**
- LOAD_IMBALANCE_DETECTED
- LOAD_IMBALANCE_ENABLED
- MAJOR_REV
- MAX_CHARGING_CURRENT_DEFAULT
- MAX_CHARGING_CURRENT_IC
- MAX_CHARGING_CURRENT_ICMAX
- MAX_CHARGING_CURRENT_RATED
- MAX_PHASE_CURRENT_BY_GRID
- MAX_TOTAL_CURRENT_BY_GRID
- METER_ENABLED
- METHOD_EN1
- MINOR_REV
- MIN_CHARGING_CURRENT
- NUMBER_OF_PHASE
- PHASES_USED
- PROHIBIT_USAGE
- SAP_NUMBER
- SERIAL_NUMBER
- SERIAL_NUMBER_INTERNAL
- SET_ICMAX
- SET_IDEFAULT
- SMART_CHARGE_ACTIVE
- SOCKET_ENABLED
- STATE
- UID
- UTMP

**Comment by toastbroat13** [comment]
wohooo... seems to be a lot of interesting data. especially the phases L1, L2 and L3 (L1_CHARGING_CURRENT and L1_USED) e.g. for double check if the wallbox switches phases (mine does not since I am using SENEC(ABL) wallboxes and have to manually switch on/off phases... ;-)

MIN_CHARGING_CURRENT also seems interesting to control if the car is limited (like when the battery is to hot...)

and what would be "SET_ICMAX" the max ampere?

thanks for your great work and hope that you will get your own wallbox asap :-)

**Comment by marq24** [comment]
The L1/L2/L3 values are currently all used - but combined into the `wallbox_power` values - but in the next release you will have all 6 (x4) Values as separate Sensors (need to be manually enabled first)

The `MIN_CHARGING_CURRENT` is available for all 4 wallboxes - so no clue what it will show/or not show - just let's give it a try...

Is it possible for you to send (by hand) a POST request with the following payload to your SENEC.Home System?
with the header `Content-Type: application/json`
```
{
"WALLBOX":{"SET_IDEFAULT": "", "SET_ICMAX ": "", "MIN_CHARGING_CURRENT": ""}
}
```
then we might get an idea what the data of these fields could meant - since currently I am a bit clueless about the UNIT's and the Measurement type

**Comment by toastbroat13** [comment]
If you tell me how to send such POSG request, I could try... ;-)

**Comment by marq24** [comment]
I assume you are running windows? have you ever worked with the power-shell command line?

If yes - make sure that you have PowerShell 7 or higher... To check:
`$PSVersionTable`

If you need to install powershell 7:
`winget install --id Microsoft.Powershell --source winget`

then you need to start this PowerShell 7 [I found it just in my windows menu] end call this functions

then you finally can execute this: YOUR-IP is obvisouly your SENEC.Home local IP 
`Invoke-WebRequest -Uri https://YOUR-IP/lala.cgi -Method POST -Headers @{"Content-type" = "application/json"} -Body '{"WALLBOX": {"SET_IDEFAULT": ""}}' -SkipCertificateCheck`

`Invoke-WebRequest -Uri https://YOUR-IP/lala.cgi -Method POST -Headers @{"Content-type" = "application/json"} -Body '{"WALLBOX": {"SET_ICMAX ": ""}}' -SkipCertificateCheck`

`Invoke-WebRequest -Uri https://YOUR-IP/lala.cgi -Method POST -Headers @{"Content-type" = "application/json"} -Body '{"WALLBOX": {"MIN_CHARGING_CURRENT": ""}}' -SkipCertificateCheck`

**Comment by toastbroat13** [comment]
Hallo Matthias.

Sorry für die späte Antwort... ich bekomme da folgendes:

PS C:\> Invoke-WebRequest -Uri https://https://192.168. XXX.XXX  //lala.cgi
-Method POST -Headers @{"Content-type" = "application/json"} -Body
'{"WALLBOX": {"SET_IDEFAULT": ""}}' -SkipCertificateCheck
Invoke-WebRequest : Es wurde kein Parameter gefunden, der dem
Parameternamen "SkipCertificateCheck" entspricht.
In Zeile:1 Zeichen:169
+ ... son"} -Body '{"WALLBOX": {"SET_IDEFAULT": ""}}' -SkipCertificateCheck
+                                                     ~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Invoke-WebRequest],
ParameterBindingException
    + FullyQualifiedErrorId :
NamedParameterNotFound,Microsoft.PowerShell.Commands.InvokeWebRequestCommand

PS C:\>
PS C:\> Invoke-WebRequest -Uri https://https://192.168.XXX.XXX//lala.cgi
-Method POST -Headers @{"Content-type" = "application/json"} -Body
'{"WALLBOX": {"SET_ICMAX ": ""}}' -SkipCertificateCheck
Invoke-WebRequest : Es wurde kein Parameter gefunden, der dem
Parameternamen "SkipCertificateCheck" entspricht.
In Zeile:1 Zeichen:167
+ ... /json"} -Body '{"WALLBOX": {"SET_ICMAX ": ""}}' -SkipCertificateCheck
+                                                     ~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Invoke-WebRequest],
ParameterBindingException
    + FullyQualifiedErrorId :
NamedParameterNotFound,Microsoft.PowerShell.Commands.InvokeWebRequestCommand

PS C:\>
PS C:\> Invoke-WebRequest -Uri https://https://192.168. XXX.XXX  //lala.cgi
-Method POST -Headers @{"Content-type" = "application/json"} -Body
'{"WALLBOX": {"MIN_CHARGING_CURRENT": ""}}' -SkipCertificateCheck
Invoke-WebRequest : Es wurde kein Parameter gefunden, der dem
Parameternamen "SkipCertificateCheck" entspricht.
In Zeile:1 Zeichen:177
+ ... ody '{"WALLBOX": {"MIN_CHARGING_CURRENT": ""}}' -SkipCertificateCheck
+                                                     ~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Invoke-WebRequest],
ParameterBindingException
    + FullyQualifiedErrorId :
NamedParameterNotFound,Microsoft.PowerShell.Commands.InvokeWebRequestCommand

Am Do., 28. Sept. 2023 um 18:00 Uhr schrieb Matthias Marquardt <
***@***.***>:

**Comment by toastbroat13** [comment]
Habs selbst gemerkt... mein Fehler ;-(

PS C:\> Invoke-WebRequest -Uri https://192.168.XXX.XXX//lala.cgi -Method
POST -Headers @{"Content-type" = "application/json"} -Body '{"WALLBOX":
{"SET_IDEFAULT": ""}}' -SkipCertificateCheck

StatusCode        : 200
StatusDescription : OK
Content           :
{"WALLBOX":{"SET_IDEFAULT":["fl_00000000","fl_00000000","fl_00000000","fl_00000000"]}}
RawContent        : HTTP/1.1 200 OK
                    Server: nginx/1.17.7
                    Date: Sun, 01 Oct 2023 11:42:42 GMT
                    Transfer-Encoding: chunked
                    Connection: keep-alive
                    Content-Type: application/json; charset=utf-8

                    {"WALLBOX":{"SET_IDEFAU…
Headers           : {[Server, System.String[]], [Date, System.String[]],
[Transfer-Encoding, System.String[]],
                    [Connection, System.String[]]…}
Images            : {}
InputFields       : {}
Links             : {}
RawContentLength  : 86
RelationLink      : {}


PS C:\>
PS C:\> Invoke-WebRequest -Uri https://192.168.XXX.XXX//lala.cgi -Method
POST -Headers @{"Content-type" = "application/json"} -Body '{"WALLBOX":
{"SET_ICMAX ": ""}}' -SkipCertificateCheck

StatusCode        : 200
StatusDescription : OK
Content           :
{"WALLBOX":{"SET_ICMAX":["fl_00000000","fl_40CCBB53","fl_00000000","fl_00000000"]}}
RawContent        : HTTP/1.1 200 OK
                    Server: nginx/1.17.7
                    Date: Sun, 01 Oct 2023 11:42:42 GMT
                    Transfer-Encoding: chunked
                    Connection: keep-alive
                    Content-Type: application/json; charset=utf-8

                    {"WALLBOX":{"SET_ICMAX"…
Headers           : {[Server, System.String[]], [Date, System.String[]],
[Transfer-Encoding, System.String[]],
                    [Connection, System.String[]]…}
Images            : {}
InputFields       : {}
Links             : {}
RawContentLength  : 83
RelationLink      : {}


PS C:\>
PS C:\> Invoke-WebRequest -Uri https://192.168.XXX.XXX//lala.cgi -Method
POST -Headers @{"Content-type" = "application/json"} -Body '{"WALLBOX":
{"MIN_CHARGING_CURRENT": ""}}' -SkipCertificateCheck

StatusCode        : 200
StatusDescription : OK
Content           :
{"WALLBOX":{"MIN_CHARGING_CURRENT":["fl_40C00000","fl_40C00000","fl_00000000","fl_00000000"]}}
RawContent        : HTTP/1.1 200 OK
                    Server: nginx/1.17.7
                    Date: Sun, 01 Oct 2023 11:42:44 GMT
                    Transfer-Encoding: chunked
                    Connection: keep-alive
                    Content-Type: application/json; charset=utf-8

                    {"WALLBOX":{"MIN_CHARGI…
Headers           : {[Server, System.String[]], [Date, System.String[]],
[Transfer-Encoding, System.String[]],
                    [Connection, System.String[]]…}
Images            : {}
InputFields       : {}
Links             : {}
RawContentLength  : 94
RelationLink      : {}

Am Do., 28. Sept. 2023 um 18:00 Uhr schrieb Matthias Marquardt <
***@***.***>:

**Comment by marq24** [comment]
super that this finally has worked - here the essential information extracted...

{"WALLBOX":{"MIN_CHARGING_CURRENT":["fl_40C00000","fl_40C00000","fl_00000000","fl_00000000"]}}
`MIN_CHARGING_CURRENT` is a float of 6.0 for both wallboxes... can we assume that this is A?

{"WALLBOX":{"SET_IDEFAULT":["fl_00000000","fl_00000000","fl_00000000","fl_00000000"]}}
so `SET_IDEFAULT` just return 0

{"WALLBOX":{"SET_ICMAX":["fl_00000000","fl_40CCBB53","fl_00000000","fl_00000000"]}}
`SET_ICMAX` for wallbox 1 is 0 for wallbox 2 is 6.397866725921631

I have to admit, that i don't understand any of the values (and the possible meaning)...

**Comment by toastbroat13** [comment]
Yes, it is Ampere.

Wallbox 2 is currently connected and charging. Wallbox 1 not.

Matthias Marquardt ***@***.***> schrieb am So., 1. Okt. 2023,
13:56:

**Comment by marq24** [comment]
you might like to observe the change of `SET_ICMAX` over time... would that be also A? but then what's the difference to the L1/2/3_CHARGING_CURRENT ?!

**Comment by toastbroat13** [comment]
I guess it is since you can set the max charging current in the app.

I am running my wallboxes with manual phase switch since solar overcome is
mainly not high enough.

Attached two pictures from the app.


Matthias Marquardt ***@***.***> schrieb am So., 1. Okt. 2023,
14:02:

**Reply by marq24** [reply]
pictures does not come through, when reply via eMail... nevertheless both values have been added... 

https://github.com/marq24/ha-senec-v3/releases/tag/3.0.7-test-006

**Comment by marq24** [comment]
@toastbroat13 since the new sensors have been added a while ago - do you already have an idea, what's the meaning/function of the SET_ICMAX value?

**Reply by toastbroat13** [reply]
@marq24 in my home assistant is no SET_ICMAX :-(
I have checked the installation for any updates but it seems to be the latest version...

**Reply by marq24** [reply]
you are on the beta channel? [and have 'activated' the new ones (they are disabled by default)]

**Comment by mschmiedel** [comment]
Hi,

I am wondering why no wallbox data is being displayed on the WebApi. Setting the Wallbox Mode and checking it would be helpful here... Just wanted to know if other than not having a wallbox yourself, did you have any reason to omit this data?

**Reply by mschmiedel** [reply]
I will add a few thoughts on this, I have understood how it is working for now:

- The configuration for the webapi is shown in the local Senec device entities. I am not 100% sure if this is wanted, but it is ok for me.
- The field senec_wallbox_1_set_icmax cannot be set, I have tested this with the senec api via postman, only senec_wallbox_1_set_icmin can be set (this entity is not being supported by the integration now). I think it would be best to add this and remove the icmax, unless other people can confirm that it works for them (this is a pity as being able to limit the current for the wallbox would be great to give the Senec battery higher priority...
- The Wallbox X Modus select does work, but only in one way: values I set are being pushed to the api, but the entity never gets a valid value. This should always be read from the api, as there is no local way of knowing this. However, these modes are not 1:1 equivalent to the api, this is composed by "chargingMode" and "compatibilityMode", maybe "allowIntercharge" if the decision is to end up adding a mode for "FAST: allow charging from Battery".

**Reply by marq24** [reply]
Since you asked for reasons "why this is currently implemented like it is"... 

As developer I prefer to release software that "just work" - therefor testing is on the the key activities BEFORE release software to the public. When I don't understand a function (here a name of a property from SENEC local interface), then I don't want that "others might find this out for me"... 

The current Senec Wallbox Integration code (combination of local API & WebAPI) have been developed based on the feedback from @iot-sle last year - MIGHT be that there are changes in the APIs in the meantime... of course this "could" be adjusted - but I have to admit, that then I would like to get feedback from @iot-sle in order to check, IF in his setup there are also the same issues as you reported - or if this might be related to hardware versions?

When there are current issues (like you describe here 'The Wallbox X Modus select does work, but only in one way: values I set are being pushed to the api, but the entity never gets a valid value'), then please be so kind and create one github issue per "bullet-point" and provide exampled (and best also postman/bruno example files (request/responses)) - so I can check, what reply from the API might not match the current code.

**Reply by mschmiedel** [reply]
Yep, I can do that, just wanted to start the discussion before creating tickets that might not be relevant for all possible setups. At the end I can only talk from the data perspective I see. For the first bullet, I don't know if this should be posted as an issue, as far as I could check the code (I do not really work with Python normally), I guess this is wanted because wallbox data is merged from local and webapi, so it might make sense to have it on the local side...

But for the other two bullet points I will create a ticket for each... ;)

**Reply by marq24** [reply]
re 'local' - to my knowledge there is no way to mark an integration as "hybrid' - The integration development started as 'local' only (for SENEC v2.1 & 3 Systems), then the V4 was released where the data could be only accessed via web... At that point of time the correct procedure would had been to release a second integration (that's web-only)... but more or less at the same time, we have realized, that even for the V3 (and v2.1) some functions can be only toggled via web - so... 

So at a certain point the local integration code checks, if also the web integration is configured - and then the two "modules" starts to interact with each other - I think there are not much other integrations around, that have this behavior (cause of Senec-SoftwareDevelopment-Style)...

**Reply by mschmiedel** [reply]
Yep, that does not seem bad, it is only confusing in some situations to know which integration will speak to device, web or both... But this could be solved in the readme If people get confused. For me it is just important to understand how things work, then I don't care what device stuff is assigned to.

**Reply by marq24** [reply]
When you have suggestions for the readme - feel free to create a PR :-D
